ghstack/config.py

```diff
@@ -130,9 +130,18 @@ def read_config(*, request_circle_token: bool = False) -> Config:  # noqa: C901
             github_username)
         write_back = True
 
-    default_branch = 'master'
     if config.has_option('ghstack', 'default_branch'):
         default_branch = config.get('ghstack', 'default_branch')
+    else:
+        default_branch = input('Default branch [master]: ')
```

Comment:
Prompting here doesn't seem like very good UX. One thing that is particularly bad here is that you're writing to user global config file by default `~/.ghstack` but the default branch is really something that varies from repository to repository.

I wonder if we should just be fetching this from the repository somehow (ideally it would be in git but I'm not sure this actually exists as a concept in git)

Reply:
Didn't see this until just now, sorry! (I can't figure out how to get notifications for this repo to go to my work email instead of my personal email.)

That makes sense to me; I'll come back to this on Tuesday and figure out a better solution. I was thinking about changing it to fetch from the repository as you mentioned; [this SO answer](https://stackoverflow.com/a/16501903/5044950) indicates that that should be easy to do using GitHub's API. Barring that, maybe I'd want to modify the config logic to allow it to use multiple files at once? (e.g. if there's both `./.ghstackrc` and `~/.ghstackrc`, then merge the two, letting settings in the former take precedence)

Reply:
Fetching from GitHub API seems most right, although it is annoying that we have to do an extra network roundtrip every execution. (I actually have this problem already for resolving repository id, so maybe you can cache that and this on a per repo basis)